### ISSUE TITLE
(sets-ext) Fatal failure at theory/sets/cardinality_extension.cpp:870

### ISSUE THREAD

**Issue by rainoftime** [CVC4/CVC4#5043]
Hi, for this formula
~~~~
(declare-const i4 Int)
(declare-fun st3 () (Set Int))
(declare-fun st7 () (Set Int))
(declare-fun st8 () (Set Int))
(declare-fun st9 () (Set Int))
(declare-fun st12 () (Set Int))
(declare-const i28 Int)
(assert (> (card st12) (abs i28)))
(assert (> (card (union (complement (intersection st7 st3)) st12)) (- (div 578 i4))))
(assert (> (card (setminus (union st9 st8) st9)) (mod (abs i28) 35)))
(check-sat)
~~~~

CVC4 commit 2786ba1
~~~~
Fatal failure within void CVC4::theory::sets::CardinalityExtension::checkNormalForm(CVC4::Node, std::vector<CVC4::NodeTemplate<true> >&) at /home/test/tofuzz/CVC4/src/theory/sets/cardinality_extension.cpp:870
Check failure

 d_im.hasSent()
~~~~

**Comment by mudathirmahgoub** [comment]
What is the cvc4 command line used to run the code above?

**Comment by rainoftime** [comment]
./cvc4 --sets-ext fname.smt2

**Comment by nafur** [comment]
This issue was apparently resolved with 1d3bb6048085342e2d85bf5193367afcd96885fa. Please come back to use if you experience this (or a similar) issue again.

**Comment by rainoftime** [comment]
The following case still triggers the issue
~~~~
(declare-fun st4 () (Set Int))
(declare-fun st12 () (Set Int))
(declare-fun st14 () (Set Int))
(assert (< (card st4) 52))
(assert (> (card (intersection (complement st12) st14)) 942))
(check-sat)
~~~~
